src/Simplex/Chat.hs

```diff
@@ -1462,7 +1492,22 @@ processAgentMessage (Just user@User {userId, profile}) agentConnId agentMessage
         _ -> pure ()
 
     processGroupMessage :: ACommand 'Agent -> Connection -> GroupInfo -> GroupMember -> m ()
-    processGroupMessage agentMsg conn gInfo@GroupInfo {groupId, localDisplayName = gName, membership, chatSettings} m = case agentMsg of
+    processGroupMessage agentMsg conn@Connection {connId} gInfo@GroupInfo {groupId, localDisplayName = gName, membership, chatSettings} m = case agentMsg of
+      INV (ACR _ cReq) ->
+        -- [async agent commands] XGrpMemIntro continuation on receiving INV
+        withCompletedCommand conn agentMsg $ \_ ->
+          case cReq of
+            groupConnReq@(CRInvitationUri _ _) -> do
+              contData <- withStore' $ \db -> do
+                setConnConnReqInv db user connId cReq
+                getXGrpMemIntroContDataGroup db user m
+              forM_ contData $ \XGrpMemIntroContData {groupMemberId, memberId, connReq, hostConnId} ->
+                forM_ connReq $ \directConnReq -> do
```

Comment:
all three (or at least two) conditions can be one case and then throw error if either is Nothing or wrong type?

Reply:
or maybe it should be no action as we are waiting for the second one